src/services/authenticationLockService.ts

```diff
@@ -48,9 +73,75 @@ export default class AuthenticationLockService {
           this.tableClient.createEntity({
             partitionKey: fiscalCode,
             rowKey: unlockCode,
+
+            // eslint-disable-next-line sort-keys
+            CreatedAt: new Date(Date.now()),
           }),
         (_) => new Error("Something went wrong creating the record")
       ),
       TE.map((_) => true as const)
     );
+
+  /**
+   * Unlock the user authentication
+   *
+   * @param fiscalCode the CF of the user
+   * @param unlockCodes the Unlock Code list to update
+   * @returns
+   */
+  public readonly unlockUserAuthentication = (
+    fiscalCode: FiscalCode,
+    unlockCodes: ReadonlyArray<UnlockCode>
+  ): TE.TaskEither<Error, true> =>
+    pipe(
+      unlockCodes,
+      ROA.map(
+        (unlockCode) =>
+          [
+            "update",
+            {
+              partitionKey: fiscalCode,
+              rowKey: unlockCode,
+              // eslint-disable-next-line sort-keys
+              Released: true,
+            },
+          ] as TransactionAction
+      ),
+      (actions) =>
+        TE.tryCatch(
+          () =>
+            // submitTransaction requires an Array
+            // eslint-disable-next-line functional/prefer-readonly-type
+            this.tableClient.submitTransaction(actions as TransactionAction[]),
+          identity
+        ),
+      TE.filterOrElseW(
+        (response) => response.status === 202,
+        () => void 0
+      ),
+      TE.mapLeft(() => new Error("Something went wrong updating the record")),
+      TE.map(() => true as const)
+    );
+
+  // -----------------------------------
+  // Private Methods
+  // -----------------------------------
+
+  private readonly getUserAuthenticationLocks = (fiscalCode: FiscalCode) => {
+    const queryFilter =
+      odata`PartitionKey eq ${fiscalCode} and not Released` as NonEmptyString;
+    return pipe(
+      this.tableClient.listEntities({
+        queryOptions: { filter: queryFilter },
+      }),
+      AI.fromAsyncIterable,
+      AI.foldTaskEither(E.toError),
+      TE.chainEitherK(
+        flow(
+          t.array(NotReleasedAuthenticationLockData).decode,
+          E.mapLeft(errorsToError)
+        )
+      )
+    );
+  };
```

Comment:
What if we use a more compact version?
The casting to NonEmptyString seems to be useful.
```suggestion
  private readonly getUserAuthenticationLocks = (fiscalCode: FiscalCode) =>
    pipe(
      this.tableClient.listEntities({
        queryOptions: {
          filter: odata`PartitionKey eq ${fiscalCode} and not Released`,
        },
      }),
      AI.fromAsyncIterable,
      AI.foldTaskEither(E.toError),
      TE.chainEitherK(
        flow(
          t.array(NotReleasedAuthenticationLockData).decode,
          E.mapLeft(errorsToError)
        )
      )
    );
```